Makefile

```diff
@@ -85,7 +85,7 @@ UNAME_M := $(shell uname -m)
 
 BUILD_IMAGE := quay.io/stackrox-io/apollo-ci:$(shell sed 's/\s*\#.*//' BUILD_IMAGE_VERSION)
 ifneq ($(UNAME_M),x86_64)
-	BUILD_IMAGE = docker.io/library/golang:1.22
+	BUILD_IMAGE = docker.io/library/golang:1.23
```

Comment:
@vladbologa perhaps this should not be hardcoded but instead retrieved (for example from `go.mod`, similar to what's being done two lines above)?
Otherwise we might have the same exact problem in a couple of months when we bump everything else to 1.24...

Reply:
https://github.com/stackrox/stackrox/pull/14385